pkg/debugapi/balances_test.go

```diff
@@ -0,0 +1,151 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package debugapi_test
+
+import (
+	"errors"
+	"github.com/ethersphere/bee/pkg/accounting/mock"
+	"github.com/ethersphere/bee/pkg/debugapi"
+	"github.com/ethersphere/bee/pkg/jsonhttp"
+	"github.com/ethersphere/bee/pkg/jsonhttp/jsonhttptest"
+	"github.com/ethersphere/bee/pkg/swarm"
+	"net/http"
+	"reflect"
+	"testing"
+)
+
+func TestBalancesOK(t *testing.T) {
+	balancesFunc := func() (ret map[string]int64, err error) {
+		ret = make(map[string]int64)
+		ret["DEAD"] = 1000000000000000000
+		ret["BEEF"] = -100000000000000000
+		ret["PARTY"] = 0
+		return ret, err
+	}
+	testServer := newTestServer(t, testServerOptions{
+		AccountingOpts: []mock.Option{mock.WithBalancesFunc(balancesFunc)},
+	})
+
+	expected := &debugapi.BalancesResponse{
+		[]debugapi.BalanceResponse{
+			{
+				Peer:    "DEAD",
+				Balance: 1000000000000000000,
+			},
+			{
+				Peer:    "BEEF",
+				Balance: -100000000000000000,
+			},
+			{
+				Peer:    "PARTY",
+				Balance: 0,
+			},
+		},
+	}
+
+	// We expect a list of items unordered by peer:
+	got := jsonhttptest.ResponseReturnDirect(t, testServer.Client, http.MethodGet, "/balances", nil, http.StatusOK, &debugapi.BalancesResponse{
+		[]debugapi.BalanceResponse{
+			{
+				Peer:    "J",
+				Balance: 5,
+			},
+		},
+	}).(*debugapi.BalancesResponse)
+
+	if !comparisonOfBalances(got, expected) {
```

Comment:
yes, it flaked because of maps unorderedness. an alternative would be 

 if fmt.Sprint(expected) != fmt.Sprint(got) {
        t.Errorf("Got: %v, not as expected: %v", got, expected)
    }

as uncovered by @paxthemax

Reply:
@janos could you please clear this up for me?

i no longer see any references to balances being sorted, although i might be missing this.

if i am correct, do we really need the `equalBalances` func instead of just doing a `reflect.DeepEqual`?

Reply:
reflect deep equal checks for order, not just content, and here the sequence of balances is not guaranteed, so you must check for content but without giving importance to order

Reply:
thanks for the clarification :+1: